Tests/ResilientDecodingTests/ResilientRawRepresentableDictionaryTests.swift

```diff
@@ -0,0 +1,149 @@
+// Created by George Leontiev on 4/23/20.
+// Copyright © 2020 Airbnb Inc. All rights reserved.
+
+import ResilientDecoding
+import XCTest
+
+private struct ResilientRawRepresentableDictionaryWrapper: Decodable {
+  @Resilient var resilientDictionary: [String: ResilientEnum]
+  @Resilient var optionalResilientDictionary: [String: ResilientEnum]?
+  @Resilient var resilientDictionaryOfFrozenType: [String: ResilientFrozenEnum]
+  @Resilient var optionalResilientDictionaryOfFrozenType: [String: ResilientFrozenEnum]?
+}
+
+final class ResilientRawRepresentableDictionaryTests: XCTestCase {
+
+  func testDecodesValidInputWithoutErrors() throws {
+    let mock = try decodeMock(ResilientRawRepresentableDictionaryWrapper.self, """
+      {
+        "resilientDictionary": { "1": "existing", "2": "existing" },
+        "optionalResilientDictionary": { "1": "existing", "2": "existing" },
+        "resilientDictionaryOfFrozenType": { "1": "existing", "2": "existing" },
+        "optionalResilientDictionaryOfFrozenType": { "1": "existing", "2": "existing" },
+      }
+      """)
+    XCTAssertEqual(mock.resilientDictionary, ["1": .existing, "2": .existing])
+    XCTAssertEqual(mock.optionalResilientDictionary, ["1": .existing, "2": .existing])
+    XCTAssertEqual(mock.resilientDictionaryOfFrozenType, ["1": .existing, "2": .existing])
+    XCTAssertEqual(mock.optionalResilientDictionaryOfFrozenType, ["1": .existing, "2": .existing])
+    #if DEBUG
+    XCTAssertEqual(mock.$resilientDictionary.errors.count, 0)
+    XCTAssertEqual(mock.$optionalResilientDictionary.errors.count, 0)
+    XCTAssertEqual(mock.$resilientDictionaryOfFrozenType.errors.count, 0)
+    XCTAssertEqual(mock.$optionalResilientDictionaryOfFrozenType.errors.count, 0)
+    #endif
+  }
+
+  func testDecodesWhenMissingKeysWithoutErrors() throws {
+    let mock = try decodeMock(ResilientRawRepresentableDictionaryWrapper.self, """
+      {
+        "resilientDictionary": { "1": "existing", "2": "existing" },
+        "resilientDictionaryOfFrozenType": { "1": "existing", "2": "existing" },
```

Comment:
Why not exclude these keys as well?

Reply:
Yup, this was based on `Array`, the test for which predated me making omission of the field not an error.

Reply:
Would it be worth updating these tests now that we've changed behavior?

Reply:
Clarified here: https://github.com/airbnb/ResilientDecoding/pull/21#discussion_r414768125